smarts/core/argoverse_map.py

```diff
@@ -0,0 +1,1127 @@
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+import heapq
+import logging
+import random
+import time
+from functools import lru_cache
+from pathlib import Path
+from typing import Dict, List, Optional, Sequence, Set, Tuple
+
+import numpy as np
+import rtree
+from cached_property import cached_property
+from shapely.geometry import Point as SPoint
+from shapely.geometry import Polygon
+
+from smarts.core.coordinates import BoundingBox, Heading, Point, Pose, RefLinePoint
+from smarts.core.lanepoints import LanePoints, LinkedLanePoint
+from smarts.core.road_map import RoadMap, RoadMapWithCaches, Waypoint
+from smarts.core.route_cache import RouteWithCache
+from smarts.core.utils.glb import make_map_glb, make_road_line_glb
+from smarts.core.utils.math import (
+    inplace_unwrap,
+    line_intersect_vectorized,
+    radians_to_vec,
+    vec_2d,
+)
+from smarts.sstudio.types import MapSpec
+
+try:
+    from av2.geometry.interpolate import interp_arc
+    from av2.map.lane_segment import LaneMarkType, LaneSegment
+    from av2.map.map_api import ArgoverseStaticMap
+except:
+    raise ImportError(
+        "You may not have installed the [argoverse] dependencies required for using Argoverse 2 maps with SMARTS. Install it first using the command `pip install -e .[argoverse]` at the source directory."
+        ""
+    )
+
+
+class ArgoverseMap(RoadMapWithCaches):
+    """A road map for an Argoverse 2 scenario."""
+
+    DEFAULT_LANE_SPEED = 16.67  # m/s
+
+    LANE_MARKINGS = frozenset(
+        {
+            LaneMarkType.DASH_SOLID_WHITE,
+            LaneMarkType.DASHED_WHITE,
+            LaneMarkType.DOUBLE_SOLID_WHITE,
+            LaneMarkType.DOUBLE_DASH_WHITE,
+            LaneMarkType.SOLID_WHITE,
+            LaneMarkType.SOLID_DASH_WHITE,
+        }
+    )
+
+    ROAD_MARKINGS = frozenset(
+        {
+            LaneMarkType.DASH_SOLID_YELLOW,
+            LaneMarkType.DASHED_YELLOW,
+            LaneMarkType.DOUBLE_SOLID_YELLOW,
+            LaneMarkType.DOUBLE_DASH_YELLOW,
+            LaneMarkType.SOLID_YELLOW,
+            LaneMarkType.SOLID_DASH_YELLOW,
+            LaneMarkType.SOLID_BLUE,
+        }
+    )
+
+    def __init__(self, map_spec: MapSpec, avm: ArgoverseStaticMap):
+        super().__init__()
+        self._log = logging.getLogger(self.__class__.__name__)
+        self._avm = avm
+        self._argoverse_scenario_id = avm.log_id
+        self._map_spec = map_spec
+        self._surfaces = dict()
+        self._lanes: Dict[str, ArgoverseMap.Lane] = dict()
+        self._roads: Dict[str, ArgoverseMap.Road] = dict()
+        self._features = dict()
+        self._lane_rtree = None
+        self._load_map_data()
+        self._waypoints_cache = ArgoverseMap._WaypointsCache()
+        if map_spec.lanepoint_spacing is not None:
```

Comment:
Yes this is a good observation. I've created a new issue for this, since it applies to all map classes. See https://github.com/huawei-noah/SMARTS/issues/1896.